src/cls/IPM/Utils/Migration.cls

```diff
@@ -0,0 +1,108 @@
+Class %IPM.Utils.Migration
+{
+
+ClassMethod RunAll(verbose As %Boolean = 1) As %Status
+{
+    Set sc = $$$OK
+    Try {
+        Do ..MigrateZPMToIPM(verbose)
+    } Catch e {
+        Set sc = e.AsStatus()
+    }
+    Quit sc
+}
+
+ClassMethod MigrateZPMToIPM(verbose As %Boolean = 1)
+{
+    If '$$$comClassDefined("%ZPM.PackageManager.Developer.Module") {
+        Write:verbose !,"Older IPM version not found; nothing to migrate.",!
+        Quit
+    }
+
+    Set sc = $$$OK
+    Set initTLevel = $TLevel
+    // Intentionally runs for current namespace only
+    Try {
+        TSTART
+
+        // ^ZPM.Dev.ModuleD -> ^IPM.Storage.ModuleD
+        Set oldId = ""
+        For {
+            Set oldId = $order(^ZPM.Dev.ModuleD(oldId),1,moduleData)
+            Quit:oldId=""
+
+            Set name = $ListGet(moduleData,2)
+            If ##class(%IPM.Storage.Module).NameExists(name) {
+                Write:verbose !,"Package already found in new storage: ",name
+                Continue
+            }
+
+            Write:verbose !,"Migrating storage for ",name," ... "
+
+            // Bring everything over - storage definitions are fortunately compatible.
```

Comment:
What if we introduce an incompatibility later and someone hasn't yet migrated? I think it would be safer to use object copy over each property. More verbose but safer in the long run

Reply:
Example of something we might do that would defeat objects-based copying: make a property private. (I want to do this for a few properties on repos.) On the other hand storage definition should retain compatibility.